src/blockchain_db/blockchain_db.cpp

```diff
@@ -26,6 +26,8 @@
 // STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF
 // THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
+#ifndef APPLE_DEBUG_BLOCKCHAIN_DB
```

Comment:
This dummy function is for preventing libblockchain_db from being empty when building it. Basically I'm moving everything from that library into libcryptonote_core. I could have directly changed CMakeLists.txt files to prevent libblockchain_db from being created at all, but that would involve a lot more lines of changes, so I opted for this #ifnded trick.